app/controllers/application_controller.rb

```diff
@@ -0,0 +1,11 @@
+class ApplicationController < ActionController::API
+  include ActionController::HttpAuthentication::Token::ControllerMethods
+
+  before_action :authenticate
+
+  def authenticate
+    authenticate_or_request_with_http_token do |token|
+      AUTHENTICATION.include?(token)
```

Comment:
In the docs for [ActionController::HttpAuthentication::Token](https://api.rubyonrails.org/classes/ActionController/HttpAuthentication/Token.html) they use `secure_compare` which sounds like worth doing to me.

Reply:
Does it make sense to use `AUTHENTICATION` as a hash here? Also, you're checking with the keys, which are the usernames.

I think what we want here is a single secret, similar to what they have in the docs linked-to above.

Reply:
Ah yes, great find. Thanks @misaka